### PR TITLE
Update titles and tags for examples with real data

### PR DESCRIPTION
**Pull request by nikolausWest**
### What
Updates the titles, and adds tags to all "python examples with real data" (a separate PR is needed for the "fake data" examples).

Renames:

| before | after |
| --- | --- |
| arkitscenes | arkit_scenes |
| colmap | structure_from_motion |
| deep_sdf | signed_distance_fields |
| dicom | dicom_mri |
| mp_pose | human_pose_tracking |
| nyud | rgbd |
| opencv_canny | live_camera_edge_detection |
| ros | ros_node |
| segment_anything | segment_anything_model |
| stable_diffusion | depth_guided_stable_diffusion |
| tracking_hf_opencv | detect_and_track_objects |

In order to minimise merge issues, this is best updated and merged AFTER:
- https://github.com/rerun-io/rerun/pull/2419
- https://github.com/rerun-io/rerun/pull/2420
- https://github.com/rerun-io/rerun/pull/2418
- #2424
- https://github.com/rerun-io/rerun/pull/2426
- #2360 

TODO

- [x] Update names script names
- [x] Update any references to the scripts
- [ ] ~~Mirror updates for Rust examples~~ (none of the renamed examples have rust counterparts)
- [x] Convert tags to list
- [x] Quick typo/grammar pass with pycharm's checker

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)

PR Build Summary: https://build.rerun.io/pr/2416

Docs preview: https://rerun.io/preview/3fb07ee/docs
Examples preview: https://rerun.io/preview/3fb07ee/examples

### CODE REVIEW

File: examples/manifest.yml [hunk 1]
```diff
@@ -100,35 +100,41 @@ root:
         - name: arkitscenes
           python: python/arkitscenes
```
Comment (emilk) [inline examples/manifest.yml:101]: This could use a better name imho

Comment (nikolausWest) [reply examples/manifest.yml:101]: Could be called "AR Scene Understanding"

Comment (abey79) [reply examples/manifest.yml:101]: Or just "AR Scene"?

Comment (abey79) [reply examples/manifest.yml:101]: "ARKit Scenes" it is (from #2420

File: scripts/build_demo_app.py [hunk 2]
```diff
@@ -239,7 +239,7 @@ def main() -> None:
         """,
         "build_args": [],
     },
-    "tracking_hf_opencv": {
+    "detect_and_track_objects": {
         "title": "Object Tracking and Segmentation",
```
Comment (emilk) [inline scripts/build_demo_app.py:243]: maybe update the title here to match?

File: scripts/run_all.py [hunk 3]
```diff
@@ -73,16 +73,16 @@ def collect_examples(fast: bool) -> list[str]:
...
-            "examples/python/deep_sdf",
-            "examples/python/dicom",
-            "examples/python/nyud",
+            "examples/python/structure_from_motion",
+            "examples/python/signed_distance_field",
+            "examples/python/dicom_mri",
+            "examples/python/rgbd",
             "examples/python/plots",
             "examples/python/raw_mesh",
             "examples/python/text_logging",
```
Comment (emilk) [inline scripts/run_all.py:82]: nit: sort

**Review by emilk — APPROVED** [review]
Much better names!

It would be nice with a table of before-and-after names in the PR description so I can learn the new names!

File: scripts/build_demo_app.py (hunk 2 above)
Comment (abey79) [reply scripts/build_demo_app.py]: done

**Comment by emilk** [comment]
The old names are still being referred to in some places. For instance, `scripts/demo_assets/static/index.html` refers to `examples/colmap` (which doesn't exist), and we are using `colmap_fiat` as a name in many places

**Pull request merged**